### ISSUE TITLE
Complains about allOf references in JSON schemas when there are none

### ISSUE THREAD

**Issue by haf** [confluentinc/ksql#6522]
**Describe the bug**
I'm running into some bug in ksqlDB that causes it to complain loudly and error the create-stream request, that I'm using allOf in a schema.

**To Reproduce**
Steps to reproduce the behavior, include:
1. The version of KSQL. 0.12.
2. Import the merged schema, like so:

```bash
#!/usr/bin/env bash
curl -L https://app.logary.tech/schemas/logary-message.merged.schema.json --silent >schema.json
echo "{}" | jq --arg schema "$(cat schema.json)" '{ schema: $schema, schemaType: "JSON" }' >tmp.json
curl -H "Accept: application/vnd.schemaregistry.v1+json, application/vnd.schemaregistry+json, application/json" -i http://localhost:8081/subjects/messages-value/versions -X POST -H "Content-Type: application/vnd.schemaregistry.v1+json" --data @tmp.json
```

3. Insert messages (simplest possible: `{ "type": "event", "event": "Hello world" }`) into a topic `messages`

4. Then in ksqldb: `create stream firehose with (kafka_topic='messages', value_format='json_sr');`

This crashes with a message 

```
[2020-10-25 15:02:11,177] INFO Processed unsuccessfully: KsqlRequest{ksql='create stream firehose with (kafka_topic='messages', value_format='json_sr');', configOverrides={}, requestProperties={}, commandSequenceNumber=Optional.empty}, reason: Unable to verify if the schema for topic messages is compatible with KSQL.
Reason: Unsupported criterion allOf for {"description":"We'll assume it's a money if it has 'amount' and 'currency' by default; but you can specify this property should you wish to.","const":"money","type":"string"}

Please see https://github.com/confluentinc/ksql/issues/ to see if this particular reason is already known.
If not, please log a new issue, including the this full error message.
https://gist.github.com/haf/1f613ffbae39d77462264752f7afbb59
```

**Expected behavior**
Not crash.

**Actual behaviour**
Since there's no `allOf` in the schema it should not crash.

**Comment by vcrfxia** [comment]
Hi @haf , thanks for reporting this. I've filed a bug report with Schema Registry, as ksqlDB uses its JSON schema converter out of the box: https://github.com/confluentinc/schema-registry/issues/1670

**Comment by vcrfxia** [comment]
This issue has been fixed in Schema Registry (see linked issue above for details), and will be picked up in a future release of ksqlDB (likely ksqlDB 0.15, not confirmed as of yet).

**Comment by hubilation** [comment]
> This issue has been fixed in Schema Registry (see linked issue above for details), and will be picked up in a future release of ksqlDB (likely ksqlDB 0.15, not confirmed as of yet).

Hello, I see this is still open and 0.15 is released. I assume this was not part of the 0.15 release?

**Comment by vcrfxia** [comment]
It was fixed as part of the 0.15 release. I forgot to close the ticket; thanks for calling this out!
